src/Sia.Connectors.Tickets/TicketProxy/ProxyConnectionInfo.cs

```diff
@@ -34,39 +45,55 @@ public ProxyConnectionInfo(string endpoint, string certThumbprint)
         /// <param name="endpoint">Proxy Endpoint</param>
         /// <param name="config">Configuration root for initialization</param>
         /// <param name="vaultName">Key vault name</param>
-        public ProxyConnectionInfo(string endpoint, KeyVaultConfiguration config, string vaultName)
-            : this(endpoint, AuthenticationType.CertificateFromKeyVault)
+        public ProxyConnectionInfo(
+            string endpoint,
+            KeyVaultConfiguration config, 
+            string vaultName
+        ) : this(
+            endpoint,
+            AuthenticationType.CertificateFromKeyVault
+        )
         {
-            _keyVault = new AzureSecretVault(config);
+            Vault = new AzureSecretVault(config);
             CertIdentifier = ThrowIf.NullOrWhiteSpace(vaultName, nameof(vaultName));
         }
 
-        protected ProxyConnectionInfo(string endpoint, AuthenticationType authType)
+        protected ProxyConnectionInfo(
+            string endpoint,
+            AuthenticationType authType
+        )
         {
             Endpoint = ThrowIf.NullOrWhiteSpace(endpoint, nameof(endpoint));
             AuthenticationType = authType;
         }
 
-        public ProxyClient GetClient() => new ProxyClient(ClientFactory.GetClient(), Endpoint);
+        public async Task<HttpClient> GetClientAsync (ILoggerFactory loggerFactory)
+        {
+            if(_client is null)
+            {
+                _client = await ClientFactory(loggerFactory).GetClientAsync();
+            }
+            return _client;
+        }
+
+        private HttpClient _client;
+        private AzureSecretVault Vault { get; }
         public AuthenticationType AuthenticationType { get; protected set; }
-        public string Endpoint { get; protected set; }
-        public string CertIdentifier { get; protected set; }
+        public string Endpoint { get; }
+        private string CertIdentifier { get; }
```

Comment:
Seems like a good candidate for a readonly field instead of a getter-only private property.

Reply:
Changed several of the getter-only properties to readonly fields.